### PR TITLE
docs: document that updateTypes: ['patch'] only works/matches if separateMinorPatch is set to true

### PR DESCRIPTION
**Pull request by casdevs**
## Changes:

update config definitions

## Context:

I noticed that `updateTypes: ['patch']` only matches patch updates if `separateMinorPatch` is set to true.
This comes somewhat unexpected and does not seem to be documented anywhere at the moment, as opposed to the documentation of the `patch` setting, where the docs correctly state `Only applies if separateMinorPatch is set to true`.

Maybe related to https://github.com/renovatebot/renovate/issues/2818

But without reading the docs, I would expect `updateTypes: ['patch']` to always match on patch updates (e.g. 1.0.0 to 1.0.1) anyway, regardless of the `separateMinorPatch` setting (which is only set if we want separate PRs or branches for patch AND minor updates)

Docs should be updated accordingly.

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added unit tests, or
- [ ] Unit tests + ran on a real repository

### CODE REVIEW

**Review by viceice — DISMISSED** [review]

**Comment by JamieMagee** [comment]
Were the docs for this already updated? Cause I'm seeing this change live at https://docs.renovatebot.com/configuration-options/#patch

File: docs/usage/configuration-options.md [hunk 1]
```diff
@@ -1268,10 +1268,12 @@ For example to apply a special label for Major updates:
 }
 ```
 
+Update type `patch` only matches if `separateMinorPatch` is set to true.
+
 ## patch
 
 Add to this object if you wish to define rules that apply only to patch updates.
-See also `major` and `minor` configuration options.
+Only applies if `separateMinorPatch` is set to true.
```
Comment (JamieMagee) [inline docs/usage/configuration-options.md:1276]: I can't see any examples where we link to anchors in the documentation, but this _should_ work, and I think it makes it easier to navigate the large documenation page.

```suggestion
Only applies if [`separateMinorPatch`](#separateMinorPatch) is set to true.
```

Comment (JamieMagee) [inline docs/usage/configuration-options.md:1276]: You don't need to update `definition.ts` _and_ `configuration-options.md`. The docs that get deployed are a combination of both. You can see by taking a look at the [patch](https://docs.renovatebot.com/configuration-options/#patch) section online, which reads:

> Configuration to apply when an update type is patch. Only applies if separateMinorPatch is set to true

File: docs/usage/configuration-options.md [hunk 2]
```diff
@@ -1268,10 +1268,12 @@ For example to apply a special label for Major updates:
 }
 ```
 
+Update type `patch` only matches if `separateMinorPatch` is set to true.
+
```
Comment (rarkins) [inline docs/usage/configuration-options.md:1272]:
```suggestion
```

File: lib/config/definitions.ts [hunk 3]
```diff
@@ -900,7 +900,7 @@ const options: RenovateOptions[] = [
   {
     name: 'updateTypes',
     description:
-      'Update types to match against (major, minor, pin, etc). Valid only within `packageRules` object.',
+      'Update types to match against (major, minor, pin, etc). Valid only within `packageRules` object. `patch` only matches if `separateMinorPatch` is set to true',
```
Comment (rarkins) [inline lib/config/definitions.ts:903]:
```suggestion
      'Update types to match against (major, minor, pin, etc). Valid only within `packageRules` object.
```

File: lib/config/definitions.ts [hunk 4]
```diff
@@ -900,7 +900,7 @@ const options: RenovateOptions[] = [
   {
     name: 'updateTypes',
     description:
-      'Update types to match against (major, minor, pin, etc). Valid only within `packageRules` object.',
+      'Update types to match against (major, minor, pin, etc). Valid only within `packageRules` object.
```
Comment (rarkins) [inline lib/config/definitions.ts:903]:
```suggestion
      'Update types to match against (major, minor, pin, etc). Valid only within `packageRules` object.',
```

**Review by JamieMagee — APPROVED** [review]

**Pull request merged**